internal/telemetry/client.go

```diff
@@ -119,8 +119,10 @@ type client struct {
 	// http.DefaultTransport and a 5 second timeout will be used.
 	Client *http.Client
 
+	// productSync syncs profiling and tracer start/stop
+	productSync sync.Mutex
 	// mu guards all of the following fields
-	mu sync.Mutex
+	mu sync.RWMutex
```

Comment:
should we change to a RW mutex to clarify which methods are meant to change client data (e.g. start) and which methods are only meant to read from client data?